mmengine/logging/logger.py

```diff
@@ -297,3 +325,38 @@ def _get_rank():
         return 0
     else:
         return get_rank()
+
+
+def _get_device_id():
+    """Get device id of current machine."""
+    try:
+        import torch
+    except ImportError:
+        return 0
+    else:
+        local_rank = int(os.getenv('LOCAL_RANK', '0'))
```

Comment:
Do all distributed launchers set this env variable? Did you test torch launcher and mpi?

Reply:
It seems that we've assumed that `LOCAL_RANK` has been set. 

https://github.com/open-mmlab/mmengine/blob/7a074fa4781ba10fee0305c24dc07ce90db15fff/mmengine/runner/runner.py#L864

I'm not sure will `mpi` launcher set this environment, but `torchrun`  and `_init_dist_slurm` will set this environment.

![image](https://user-images.githubusercontent.com/57566630/224220080-618c0b89-fb7a-4e57-b172-dae14d74628e.png)


![image](https://user-images.githubusercontent.com/57566630/224220227-0a4419fb-6ba6-4f4c-8cf3-15a2e43ce7b8.png)